src/lib/php/UI/template/components/login-form.html.twig

```diff
@@ -41,6 +47,17 @@
         {% endif %}
       </div>
     </div>
+    <script>
+        const togglePassword = document.querySelector('#togglePassword');
+        const password = document.querySelector('#pwd');
+        togglePassword.addEventListener('click', function (e) {
+          // toggle the type attribute
+          const type = password.getAttribute('type') === 'password' ? 'text' : 'password';
+          password.setAttribute('type', type);
+          // toggle the eye slash icon
+          this.classList.toggle('fa-eye-slash');
+      });
+    </script>
```

Comment:
The script is repeated at 3 locations. Please add it to an appropriate JS file under `src/www/ui/scripts/` or create a new file and import it in the twig.

You can check the parent twig file by searching the current file name.
E.g.:
```bash
$ grep -rn 'login-form.html.twig' src/
src/www/ui/template/home.html.twig:11:    {% include 'components/login-form.html.twig' with {
src/www/ui/template/login.html.twig:13:    {% include "components/login-form.html.twig" with {
```

It tells the file is imported via `home.html.twig` and `login.html.twig`.

Find an appropriate place to import this script file.

Reply:
> The script is repeated at 3 locations. Please add it to an appropriate JS file under `src/www/ui/scripts/` or create a new file and import it in the twig.
> 
> You can check the parent twig file by searching the current file name. E.g.:
> 
> ```shell
> $ grep -rn 'login-form.html.twig' src/
> src/www/ui/template/home.html.twig:11:    {% include 'components/login-form.html.twig' with {
> src/www/ui/template/login.html.twig:13:    {% include "components/login-form.html.twig" with {
> ```
> 
> It tells the file is imported via `home.html.twig` and `login.html.twig`.
> 
> Find an appropriate place to import this script file.
Thank you for the input .. I have a small doubt the script for login and [edit, add] differs in two parameters because both the parameters are carrying there own css so I should be calling the same method twice with the different parameters
in the js?